lua/dap/session.lua

```diff
@@ -560,17 +560,31 @@ function Session:handle_body(body)
         log.warn('No event handler for ', decoded)
       end
     end)
-  elseif decoded.type == 'request' and decoded.command == 'runInTerminal' then
-    self:run_in_terminal(decoded)
+  elseif decoded.type == 'request' then
+    local handler = self.handlers.reverse_requests[decoded.command]
+    if handler then
+      handler(self, decoded)
+    end
+    log.warn('No handler for reverse request', decoded)
   else
     log.warn('Received unexpected message', decoded)
   end
 end
 
 
-local function session_defaults(opts)
+local default_reverse_request_handlers = {
+  runInTerminal = run_in_terminal
+}
+
+
+local function session_defaults(adapter, opts)
   local handlers = {}
   handlers.after = opts.after
+  handlers.reverse_requests = vim.tbl_extend(
+    'keep',
```

Comment:
so this means that the default wins? And if the adapter has it as well, it will not be overridden?

Should it maybe be `'error'` such that if the adapter overrides, an error is thrown instead of it being silently swallowed? This will hopefully make it a bit more clear that something is off (after an upgrade). Potentially we could even catch the error here and throw it with a clear error message that the adapter is overriding a handler it cannot override (because there is now a default)

Reply:
Good point, I'll change it to error